src/env.mjs

```diff
@@ -33,6 +33,7 @@ export const env = createEnv({
     // NEXT_PUBLIC_CLIENTVAR: z.string().min(1),
     NEXT_PUBLIC_SUPABASE_ANON_KEY: z.string(),
     NEXT_PUBLIC_SUPABASE_URL: z.string(),
+    NEXT_PUBLIC_SUPPORTED_SCH_DOMAINS: z.string(),
```

Comment:
This can indeed be useful. Adopted the changes at [75d5a3e](https://github.com/AfterClass-io/afterclass.io-v2/pull/93/commits/75d5a3efddd78fa01d9c48d56d2afcd6db78a658)

Note: one issue with this is that even though it performs fqdn check, developers will not be able check if the env domains are indeed the ones needed (seems no way to handle)

Reply:
thanks for the changes and the note!


> Note: one issue with this is that even though it performs fqdn check, developers will not be able check if the env domains are indeed the ones needed (seems no way to handle)

hmm, i don't think that falls under the responsibly of this env schema - to check what the actual correct env var values to use are, especially since this would probably be unsolveable with just zod 😅

my thoughts:

i'd think it would be fair to assume that devs should be aware that the use of this env var is to declare which email domain we support and list the correct ones accordingly

the purpose of the schema validation is not really to enforce devs to list the correct domains (there's probably no automated way to do that) but rather, to make sure that the domains listed in the env var at `.env` is written in the correct format - `XXX="x.com,y.com,z.com"`, rather than something like `XXX="x.com, y.com, z.com"`  (space after each comma) or something more ambiguous like `XXX="['x.com','y.com','z.com']"` (maybe wanted to use `JSON.parse` to read?)

with just `z.string()`, all the above listed erroneous formats are accepted and may be unknowningly consumed in the wrong way downstream (`"['x.com','y.com','z.com']".split(",")` for example)

this transformation and validation also makes it more specific to what the structure of the env var is, which is an array of fqdns, parsed from a format like `XXX="x.com,y.com,z.com"`

Reply:
75d5a3efddd78fa01d9c48d56d2afcd6db78a658 LGTM, feel free to resolve this review comment if there are no other issues!